### PR TITLE
Decaffeination

### PR DESCRIPTION
**Pull request by mserranom**
### Description

Project decaffeination -- uses the decaffeinate script with fixes from https://github.com/overleaf/dev-environment/pull/354

#### Potential Impact

Shouldn't have any, but keeping an eye post-deployment is needed, just in case.

#### Manual Testing Performed

- [x] run unit/acceptance tests
- [x] test dev-environment

#### Deployment Checklist

- [ ] Test staging
- [ ] Test production
- [ ] Monitor production dashboard and logs

### CODE REVIEW

File: app/js/RoomManager.js [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+        result.push(this.leaveEntity(client, entity, id))
+      }
+      return result
+    })()
+  },
+
+  emitOnCompletion(promiseList, eventName) {
+    const result = Promise.all(promiseList)
+    result.then(() => RoomEvents.emit(eventName))
+    return result.catch((err) => RoomEvents.emit(eventName, err))
```
Comment (das7pad) [inline app/js/RoomManager.js:74]: _This is not from latest master._

File: app/js/WebApiManager.js [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+            return callback(err)
+          }
+          return callback(
+            null,
+            data.project,
+            data.privilegeLevel,
+            data.isRestrictedUser
+          )
+        } else if (response.statusCode === 429) {
+          logger.log(project_id, user_id, 'rate-limit hit when joining project')
```
Comment (das7pad) [inline app/js/WebApiManager.js:69]: Side note: we should wrap the context in an object.

File: app/js/WebsocketLoadBalancer.js [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+
+  emitToAll(message, ...payload) {
+    return this.emitToRoom('all', message, ...Array.from(payload))
+  },
+
+  listenForEditorEvents(io) {
+    logger.log(
+      { rclients: this.rclientPubList.length },
+      'publishing editor events'
+    )
```
Comment (das7pad) [inline app/js/WebsocketLoadBalancer.js:73]: This duplicate already existed in coffee-script.

File: buildscript.txt [hunk 4]
```diff
@@ -1,10 +1,10 @@
 real-time
---public-repo=True
---language=coffeescript
---env-add=
---node-version=10.21.0
 --acceptance-creds=None
---dependencies=redis
+--dependencies=None
```
Comment (das7pad) [inline buildscript.txt:3]: Please restore the old buildscript.txt file. The current revision is missing `public-repo=True` and `dependencies=redis`. CI is marked as success but in reality it failed.

File: package.json [hunk 5]
```diff
@@ -38,13 +36,31 @@
...
+    "mocha": "^4.0.1",
+    "prettier": "^2.0.5",
+    "prettier-eslint-cli": "^5.0.0",
     "sandboxed-module": "~0.3.0",
     "sinon": "^2.4.1",
-    "mocha": "^4.0.1",
-    "uid-safe": "^1.0.1",
-    "timekeeper": "0.0.4"
+    "timekeeper": "0.0.4",
+    "uid-safe": "^1.0.1"
```
Comment (das7pad) [inline package.json:64]: Can we rebase this PR onto the dependencies upgrade -- there is a conflict with `uid-safe`.

**Comment by mserranom** [comment]
Thanks for your feedback @das7pad I think it's easier sand cleaner to close this PR to address the issues you underlined separately. Just the rebase with `master` and the dependency updates is complex enough.

File: app/js/RoomManager.js (hunk 1 above)
Comment (mserranom) [reply app/js/RoomManager.js:74]: I would have expected `make all` would pull the latest changes :/

**Comment by das7pad** [comment]
When you force-push in here it would be much easier for me to review again (Github will display a partial diff).

**Comment by mserranom** [comment]
> When you force-push in here it would be much easier for me to review again (Github will display a partial diff).

OK, can give it a try

File: app/js/WebApiManager.js (hunk 2 above)
Comment (das7pad) [reply app/js/WebApiManager.js:69]: Just to be clear, there is no need to fix this in the scope of this PR :)

File: app/js/WebsocketLoadBalancer.js (hunk 3 above)
Comment (das7pad) [reply app/js/WebsocketLoadBalancer.js:73]: Just to be clear, there is no need to fix this in the scope of this PR :)

**Pull request closed without merging**